### DISCUSSION TITLE
Meeting notes #12, 2023-03-20

### DISCUSSION THREAD

**Discussion by jongfeel** [fkdl0048/Mentoring-Project#55] [General]
미리 질문 주신 내용에 대해 답변한 걸 기록해 봅니다.

- #54 

gitflow의 간략한 버전을 채택해서 진행하면 좋을 수 있습니다. 특정 파일에 대한 branch를 따로 두는 건 merge 작업을 계속 해야 하는 부담감이 있고 실수로 merge를 안하는 경우가 있을 수 있으므로 추천하지는 않습니다.
develop를 기준으로 release를 하면 main branch로 merge 하고
develop에서 각 작업 branch를 만들고 작업이 되면 develop로 pull request를 통해 코드리뷰를 하고 develop가 항상 빌드가 되는 상태를 유지시키는게 좋습니다.

- #53 

### branch checkout에 대해서

작업 단위로 branch를 만들고 그걸 빠른 시간 내에 develop으로 merge 해 나가는 방식으로 진행하면 좋습니다.
이후 또 작업이 필요하거나 버그를 수정한다고 하면 거기에 맞는 branch가 생성되고 작업을 하는게 맞겠죠

### book club

책 읽는 모임도 자율적으로 하게 하고 부담을 주지 않는 선에서 진행하면 좋습니다.
보통 하루나 이틀이면 읽을 수 있는 분량으로 하고 논의 주제를 가져오게 하면 모임이 진행될수록 책을 읽고 와야 한다는 생각이 들게 되어 있습니다.
중간에 사정이 있어서 빠지는 것도 자유롭게 하시고, 준비를 못해서 모임에 빠지는 경우도 논의 주제를 준비해서 언제 pull request를 줄 수 있는지 물어보고 그때 해달라고 하시면 됩니다. 그것 마저도 못하면 게으르거나 바쁜 사람이므로 자연스럽게 도태되거나 안하게 됩니다.

### 기획에게 논리적 사고력이 필요한가

필요합니다. 

반드시 비슷한 게임을 하게 하고 그 게임의 규칙에 대해 설명해 보라고 한 다음에 그걸 기획서로 만들면 어떻게 만들면 좋을지 해보라고 하세요. 그러면 여태까지 만들었던 기획서는 구멍이 많다는 걸 깨닫게 됩니다.

또 샌드위치 만들기 동영상도 같이 한번씩 보시면 왜 디테일한 규칙을 만들어 놓지 않으면 뭔가를 만들 수 없는 지를 소프트웨어의 비유가 아니더라도 일상 생활에서도 깨달을 수 있습니다.

샌드위치 코딩이라는 영상인데 프로그래밍 모르는 사람이 보면 좋은 영상입니다.
https://youtu.be/I5cq54MFQCo

### 스스로 설정한 목표에 대한 일정 추청

어렵습니다. 그래서 거의 시간 단위로 추정 가능한 작업 단위로 나누고 그걸 합쳐서 일단위, 주단위 작업 완료 목표를 세워야 뭔가 만들고 있다는 느낌을 많이 받을 수 있습니다. 일정이 변경 되는 건 허용을 해줘도 됩니다. 스스로 능동적이고 프로페셔널한 작업을 하고 있는지 생각할 시간을 가져보는 것도 좋습니다.

미하이 칙센트미하이는 몰입을 통해 스스로 세운 목표를 달성하는 것이야 말로 엄청난 성과를 얻을 수 있다고 했습니다.

### github project 관리

작업 내용과 코드 내용이 일치하면 잘 진행하고 있다는 뜻입니다. 작업 내용에 없는 코드를 추가로 작성했다면 작업 내용 문서에 반영하는 식으로 진행하면 관리가 어느 정도 됩니다.

### CI/CD

초기에 세팅하는걸 추천합니다. 자동화가 어렵다면 수동으로라도 매뉴얼을 만들어서 진행해보고 반복적으로 시간을 쓰면서 하고 있다는 깨달음을 얻으면 그 때 자동화 방법을 찾아서 적용하면 좋습니다.

**Comment by fkdl0048** [comment]
감사합니다..!
